### ISSUE TITLE
Eve Light Switch History?

### ISSUE THREAD

**Issue by grover** [simont77/fakegato-history#42]
Hi,

Elgato sells the Eve Light Switch in the US and interestingly it seems to offer a history for Characteristic.On. As I'm in Germany I can't use this out of the box, however it would be great to know if support for this is underway?

I'd be  looking to add this to my automation switches. Incidentally if I try to expose the history for the switches, the app already shows an On History, but doesn't show any data - so the data format must differ from the others somehow.

Thanks!

**Comment by simont77** [comment]
In principle it is possible and should not be difficult, but I need a dump of characteristics 116 and 117. So we need someone with a real Eve Switch able to follow the procedure described in the readme.

**Comment by NebzHB** [comment]
i would love that too, i'm not good with electricity but i guess someone could modulate 220V -> 120V to make it work ...

**Comment by grover** [comment]
That's what I'm currently contemplating: Purchasing one along with the necessary voltage converter to try to reverse engineer it. It shouldn't be too difficult given that I'd pair it with homebridge-ranger and just sniff the traffic along the way.

**Comment by tobekas** [comment]
Any news on this?
I tried adding the 'motion' type history for a switch service. As a result, the History dropdown actually appears in the Eve App, but its always showing "No data available".
(This is the same behaviour as with the 'door' sensor history, if the optional characteristics OpenDuration / OpenDuration are not provided)
I suspect that there is also an optional characteristic missing, that is necessary to show the switching-event history.
Is there anyone who owns a Eve Light Switch, and can investigate?

**Comment by simont77** [comment]
No, unfortunately I did not receive any communication dump yet.

**Comment by tobekas** [comment]
@simont77: Great news: Since #75 I found out a very simple way to get the switch history (on/off) working:

Just add `TYPE_SWITCH = 'switch';`

with 
```
case TYPE_SWITCH:
   this.accessoryType116 = "01 0e01";
   this.accessoryType117 = "01";
```
(rest of the code can be copy-pasted from motion and just add `case TYPE_SWITCH` to the further switch-cases for TYPE_DOOR/TYPE_MOTION.)

And voila: History appears (unfortunately only in combination with the Switch-service. In my test it does not appear with Lightbulb or Outlet, but this may be related to Eve's stricter requirements discussed in #76). Here I am gladly open for more information from the community.

**Comment by tobekas** [comment]
Since Eve light switch is coming to most EU countries next month, this would be a great extension.
There will also be a schedule-feature... but I have no plan for this (probably similar to the Thermo-schedule).
Luckily, switch history also appears without this schedule in v3.8.1

**Comment by tobekas** [comment]
@simont77: Have you already tested my suggested characteristics 116 and 117 for this topic?

**Comment by simont77** [comment]
Sorry, I don't have so much time at the moment, but feel free to submit a PR, and I will try to merge it.

**Comment by tobekas** [comment]
Great, it's working fine!
I tested with homebridge-mqttthing plugin.

I think this issue can be closed...
